staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -168,32 +168,6 @@ func TestWatchErrorEventIsBlockingFurtherEvent(t *testing.T) {
 // impact.
 // =======================================================================
 
-func TestWatchErrResultNotBlockAfterCancel(t *testing.T) {
-	origCtx, store, _ := testSetup(t)
-	ctx, cancel := context.WithCancel(origCtx)
-	w := store.watcher.createWatchChan(ctx, "/abc", 0, false, false, storage.Everything)
-	// make resultChan and errChan blocking to ensure ordering.
-	w.resultChan = make(chan watch.Event)
-	// The event flow goes like:
-	// - first we send an error, it should block on resultChan.
-	// - Then we cancel ctx. The blocking on resultChan should be freed up
-	//   and run() goroutine should return.
-	var wg sync.WaitGroup
-	wg.Add(1)
-	go func() {
-		w.run(false, true)
-		wg.Done()
-	}()
-	wg.Add(1)
-	go func() {
-		w.sendError(fmt.Errorf("some error"))
```

Comment:
This is the line that was causing issues in the reported bug.
The problem was happening if sendError was effectively called after resultChan was closed (which could happen because we're calling `cancel()` in the line below.

With `sendError()` being now a synchronous method, I think this test now doesn't make sense.
But if you have suggestions for different tests, I'm happy to hear.

Reply:
yeah... we don't have errChan any more, and all sends to resultChan are easily inspectable to also select on ctx.Done() ... agree this test isn't really relevant any more